src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -896,6 +897,39 @@ public void Close()
             }
         }
 
+        public void Kill(bool entireProcessTree)
+        {
+            if (!entireProcessTree)
+            {
+                Kill();
+            }
+            else
+            {
+                // Ensures that an InvalidOperationException is thrown if the process hasn't started yet -- present to mimic the behavior of Kill()
+                EnsureState(State.HaveId);
+
+                KillTree();
+            }
+        }
+
+        private void KillChildren(IReadOnlyList<Process> children)
+        {
+            try
+            {
+                foreach (Process childProcess in children)
+                {
+                    childProcess.KillTree();
```

Comment:
Shouldn't this make the attempt for each of the children (i.e. `try ... catch` around `childProcess.KillTree()`)?

Reply:
Actually it appears that it never throws, at least in the Windows case? Which begs the question: why the `finally`?

Reply:
There are a couple points in `GetChildProcesses()` (called by `KillTree()`) that could throw. If this happens, the `finally` on the outside of the `foreach` ensures that all `Process` instances in `children` are disposed. 

If the `try...finally` were moved to immediately around `childProcess.KillTree()` and that calls throws, the current `childProcess` will be disposed but any left waiting for iteration will be undisposed.